### PR TITLE
Fix/unit for diff based indices

### LINKED ISSUE #255: BUG: Unit handling for temperature difference indices
* icclim version: 6.1.3
* Python version: 3.10.8

### Description
It appears that for indices that involve calculating the difference between two temperatures (e.g. DTR, ETR, VDTR), unit conversion is performed after the difference is taken. This is problematic if the input units are Kelvin, because the difference between two temperatures in Kelvin is equivalent to the difference in Celsius and hence no unit conversion is required.

### Minimal reproducible example
[Example 2: Index ETR](https://icclim.readthedocs.io/en/stable/how_to/recipes_ecad.html#example-2-index-etr)
```python
files_tasmax = [
    "tasmax_day_CNRM-CM5_historical_r1i1p1_19300101-19341231.nc",
    "tasmax_day_CNRM-CM5_historical_r1i1p1_19350101-19391231.nc",
]
files_tasmin = [
    "tasmin_day_CNRM-CM5_historical_r1i1p1_19300101-19341231.nc",
    "tasmin_day_CNRM-CM5_historical_r1i1p1_19350101-19391231.nc",
]

out_f = "ETR_year_CNRM-CM5_historical_r1i1p1_1930-1939.nc"  # OUTPUT FILE: annual values of ETR

icclim.index(
    index_name="ETR",
    in_files=[files_tasmax, files_tasmin],
    var_name=["tasmax", "tasmin"],
    slice_mode="year",
    out_file=out_f,
)
```

### Output received
The output (in `out_f`) is a data array (with units = C) whose values range from -270 to -182 because unit conversion (i.e. subtracting 273.15 to move from K to C) appears to have been performed after the difference in temperatures (in Kelvin) was calculated.

```python
import xarray as xr

ds = xr.open_dataset('ETR_year_CNRM-CM5_histrorical_r1i1p1_1930-1939.nc')

data_min = ds['ETR'].values.min()
data_max = ds['ETR'].values.max()
data_units = ds['ETR'].attrs['units']

print('Data minimum:', data_min)
print('Data maximum:', data_max)
print('Data units:', data_units)
```

```
Data minimum: -270.46213
Data maximum: -182.72186
Data units: C
```

### PR DESCRIPTION
**Pull request by bzah**
### Pull Request to resolve #255
- [x] Unit tests cover the changes.
- [ ] These changes were tested on real data.
- [ ] The relevant documentation has been added or updated.
- [x] A short description of the changes has been added to `doc/source/references/release_notes.rst`.

### Describe the changes you made
Indicators computing the difference between two variables must first convert the units of input variables to the expected output unit in order to avoid converting the output which would be a relative "temperature".
In other words: a 15 Kelvin difference *is* equivalent to a 15 degC and *is not* to a -258.15 degC.

### CODE REVIEW

**Review by pagecp — APPROVED** [review]
I think it is the proper way to implement this fix!
